src/ai/backend/manager/registry.py

```diff
@@ -1061,16 +1060,13 @@ async def enqueue_session(
             )
 
             # Shared memory.
-            # We need to subtract the amount of shared memory from the memory limit of
-            # a container, since tmpfs including /dev/shm uses host-side kernel memory
-            # and cgroup's memory limit does not apply.
-            shmem = resource_opts.get("shmem", None)
-            if shmem is None:
-                shmem = labels.get("ai.backend.resource.preferred.shmem", "64m")
-            shmem = BinarySize.from_str(shmem)
+            # Do not including the shared memory when comparing the memory slot with image requiring resource slot.
+            shmem = BinarySize.from_str(
+                resource_opts.get("shmem")
+                or labels.get("ai.backend.resource.preferred.shmem")
+                or DEFAULT_SHARED_MEMORY_SIZE
```

Comment:
I think injection of the default value should be taken care of by the client side, and there should be a way to set `shmem` to zero intentionally when required. If we could distinguish "undefined" and "zero" here, setting the default value for "undefined" case would be fine.